Smart wallet react starter

jmderby opened the pull request:
## Description

This PR focuses on porting `scw_demo` into the SDK monorepo. The primary objective was to ensure seamless integration with the Lerna monorepo setup, particularly making sure that the build and start processes are fully compatible. To achieve this, several adjustments were made, including:

- Lerna Compatibility: Modifications to ensure that scw_demo works harmoniously within the Lerna monorepo, focusing on build and start script compatibility.
- tsconfig Clean-up: Streamlined the TypeScript configuration to enhance maintainability and performance, removing unnecessary or redundant configurations.

## Test plan

- Ensure original wallet creation flow still works.

devlyn37 reviewed (commented):
Left some suggestions, also noticing that the build is failing but it looks unrelated to anything added here. Do you know the story there?

MatiNavCrossmint commented on apps/wallets/aa/react-starter-demo/src/app/utils/createAAWallet/createW3AAAWallet.ts:
```diff
@@ -0,0 +1,53 @@
+
+    const { email } = parseToken(jwt);
+
+    console.log("Here they are:");
+    console.log(process.env.REACT_APP_CROSSMINT_API_KEY_PROD);
+    console.log(process.env.REACT_APP_CROSSMINT_API_KEY_STG);
```
We should remove these ones

MatiNavCrossmint commented on apps/wallets/aa/react-starter-demo/src/app/utils/mintApi.ts:
```diff
@@ -0,0 +1,132 @@
+import { getERC20ContractAddress, getNFTContractAddress } from "./getContracts";
+
+export const transferTokenERC20 = async (account: EVMSmartWallet, to: string, rawAmount: string) => {
+    console.log("Transfer Token ERC20", account.address);
```
this one too

MatiNavCrossmint reviewed (commented):
Amazing, thanks for doing this!

Quick question, as it has a big amount of changes, it is pretty difficult to track them. So, is this all about moving the files without needing to modify them, or did you need to make some changes? I ask mostly about the ones that run the react app, not the ones for building or, for example, the package.json

mPaella commented on package.json:
```diff
@@ -20,6 +20,7 @@
         "prettier:check": "prettier --check \"{*,**/*}.{js,ts,jsx,tsx,json,mdx}\" --ignore-unknown",
         "prettier:write": "prettier --write \"{*,**/*}.{js,ts,jsx,tsx,json,mdx}\" --ignore-unknown",
         "publish": "lerna publish from-package",
+        "start:react-starter-demo": "pnpm --filter @crossmint/react-starter-demo run start",
```
i'd rather not add this script to the top level package.json

jmderby replied:
You bet, i'll remove it :+1:

jmderby commented:
> Amazing, thanks for doing this!
> 
> Quick question, as it has a big amount of changes, it is pretty difficult to track them. So, is this all about moving the files without needing to modify them, or did you need to make some changes? I ask mostly about the ones that run the react app, not the ones for building or, for example, the package.json

Thanks @MatiNavCrossmint.
Great question, I tried to modify as little as possible for a seamless port over. I mainly was having to modify minor things like the tsconfig, I removed an instance of a type for next/image being imported in a component, I added sensitive vars to the .env file, tweaked a little bit of the eslintrc in order for the errors to not being throwing, formatted the files with using the configured lerna linter and prettier. Those aside I really just moved stuff over.
